db/push.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	nomsjson "roci.dev/diff-server/util/noms/json"
+)
+
+// These request/response types might want to live in a different location, eg 
+// in diff-server/serve/types.
+type Mutation struct {
+	ID   uint64          `json:"id"`
+	Name string          `json:"name"`
+	Args json.RawMessage `json:"args"`
+}
+
+type BatchPushRequest struct {
```

Comment:
Nit: do you mind reorganizing these from top down, I think it's easier to read. Like start at BatchPushRequest and its children, then BatchPushResponse and its?